### PR TITLE
Unity Container Version Lock Support

### PR DESCRIPTION
**Pull request by rajinir**

### CODE REVIEW

**Comment by rajinir** [comment]
Pep8 errors are not related to this patch and they are already fixed in the other pending patch

File: src/deploy/osp_deployer/director.py [hunk 1]
```diff
@@ -161,6 +161,19 @@ def upload_update_conf_files(self):
             dest_file = self.home_dir + yaml
             self.upload_file(source_file, dest_file)
 
+            unity_lock = "/unity_container_lock.ini"
```
Comment (jtwill98) [inline src/deploy/osp_deployer/director.py]: As a developer that has ported code to many operating systems, I'm not a fan of this as the directory separator "/" is hard coded to the file name which makes porting a nightmare.  I'm only going to comment here as an FYI.

File: src/deploy/osp_deployer/director.py [hunk 2]
```diff
@@ -161,6 +161,19 @@ def upload_update_conf_files(self):
...
+            unity_lock_file = self.settings.lock_files_dir + unity_lock
+            if self.settings.enable_unity_backend is True:
+               cmd = "grep unity_cinder__container_version " + unity_lock_file +
+                     " | awk -F '=' '{print $2}'"
+               self.settings.cinder_unity_container_version = self.run_tty(cmd)
+            if self.settings.enable_unity_manila_backend is True:
+               cmd = "grep unity_manila_container_version " + unity_lock_file +
```
Comment (gaelrehault) [inline src/deploy/osp_deployer/director.py]: Shouldn t this be _**manila_unity_container_version**_

File: src/deploy/osp_deployer/director.py [hunk 3]
```diff
@@ -161,6 +161,19 @@ def upload_update_conf_files(self):
             dest_file = self.home_dir + yaml
             self.upload_file(source_file, dest_file)
 
+            unity_lock = "/unity_container_lock.ini"
+            unity_lock_file = self.settings.lock_files_dir + unity_lock
+            if self.settings.enable_unity_backend is True:
+               cmd = "grep unity_cinder__container_version " + unity_lock_file +
```
Comment (gaelrehault) [inline src/deploy/osp_deployer/director.py]: Shouldn t this be grep **_cinder_unity_container_version_**

**Comment by gaelrehault** [comment]
Besides the errors above that wouldn t apply the versions properly, I dont see any logic in there to apply those versions based on the value of the ini _if_ version locked is enabled. This looks to me that those values would always be picked up, in which case they could just be set in the .ini . 
I think this PR should be reverted and the issues adressed.

File: src/deploy/osp_deployer/director.py (hunk 3 above)
Comment (rajinir) [reply src/deploy/osp_deployer/director.py]: You are right, I have to fix this. Good catch

**Pull request merged**